stl/inc/mutex

```diff
@@ -601,16 +588,21 @@ public:
 #if _HAS_CXX20
         static_assert(chrono::is_clock_v<_Clock>, "Clock type required");
 #endif // _HAS_CXX20
+        if (!_Mtx_current_owns(_Lck.mutex()->_Mymtx())) {
+            _Throw_Cpp_error(_OPERATION_NOT_PERMITTED);
+        }
         for (;;) {
             const auto _Now = _Clock::now();
             if (_Abs_time <= _Now) {
                 return cv_status::timeout;

```

Comment:
```suggestion
                // we don't lock-and-relock _Lck for this case because it's not observable
                return cv_status::timeout;
```

Reply:
Hmm, should it be `unlock-and-relock`?

Reply:
Derp, yes it should.